service/index/storage.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package index
+
+import (
+	"github.com/dgraph-io/badger/v2"
+
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type Storage interface {
```

Comment:
Since you already use `storage.NewLibrary`, and you use `ReadLibrary` and `WriteLibrary`, shouldn't the interface be called `Library` here?

Reply:
From the POV of `index`, having a dependency to something just called `Library` is too vague IMO. `ReadLibrary` and `WriteLibrary` lose all their context without the package name (`storage`). From the index, I feel like we should only deal with `storage` and it doesn't matter whether it's a library or not, it's just the storage interface.

From the perspective of the `storage` package however, I agree that using the _library_ naming makes sense, since it makes it clear what the purpose of the struct is.